### PR TITLE
Fix printing of instantiation patterns

### PR DESCRIPTION
**Pull request by ajreynol**
Previously we were assuming all children of INST_PATTERN_LIST were of kind INST_ATTRIBUTE or INST_PATTERN, which is not the case.

### CODE REVIEW

File: src/printer/smt2/smt2_printer.cpp [hunk 1]
```diff
@@ -862,7 +862,8 @@ void Smt2Printer::toStream(std::ostream& out,
     out << ')';
     return;
   }
-  case kind::INST_PATTERN: break;
+  case kind::INST_PATTERN:
+  case kind::INST_NO_PATTERN: break;
```
Comment (4tXJ7f) [inline src/printer/smt2/smt2_printer.cpp]: This case shouldn't really occur because of the `nc[0]` below, no?

Comment (ajreynol) [reply src/printer/smt2/smt2_printer.cpp]: That's right, although the kind should appear here regardless.

Comment (4tXJ7f) [reply src/printer/smt2/smt2_printer.cpp]: Should we maybe make it `Unreachable()` in that case?

Comment (ajreynol) [reply src/printer/smt2/smt2_printer.cpp]: No, we still should be able to print it out internally.  It's just that it's main use case (as a child of forall) forgoes printing it.

Comment (4tXJ7f) [reply src/printer/smt2/smt2_printer.cpp]: That's fair.

**Pull request merged**